master/internal/task/idler/watcher_test.go

```diff
@@ -0,0 +1,42 @@
+package idler
+
+import (
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/determined-ai/determined/master/internal/sproto"
+)
+
+func TestIdleTimeoutWatcherUseRunnerState(t *testing.T) {
+	TickInterval = 10 * time.Millisecond
+	var actionDone atomic.Bool
+	timeout := time.Second
+	cfg := sproto.IdleTimeoutConfig{
+		ServiceID:       "test",
+		TimeoutDuration: timeout,
+		UseRunnerState:  true,
+	}
+
+	Register(cfg.ServiceID, &cfg, func(error) {
+		actionDone.Store(true)
+	})
+	defer Unregister(cfg.ServiceID)
+
+	RecordActivity(cfg.ServiceID)
+
+	waitForCondition(t, 10*timeout, actionDone.Load)
+}
+
+func waitForCondition(
+	t *testing.T,
```

Comment:
This doesn't use the `T`; is it supposed to be testing something?

Reply:
had logging in initially